src/main/java/edu/kis/powp/jobs2d/command/gui/CommandManagerWindow.java

```diff
@@ -64,13 +67,36 @@ public CommandManagerWindow(DriverCommandManager commandManager) {
 		c.weighty = 1;
 		content.add(btnClearCommand, c);
 
+
 		JButton btnClearObservers = new JButton("Delete observers");
 		btnClearObservers.addActionListener((ActionEvent e) -> this.deleteObservers());
 		c.fill = GridBagConstraints.BOTH;
 		c.weightx = 1;
 		c.gridx = 0;
 		c.weighty = 1;
 		content.add(btnClearObservers, c);
+
+		DefaultListModel model = new DefaultListModel();
+		CommandHistory.setListModel(model);
+		JList commandHistoryList = new JList(model);
+		commandHistoryList.setSelectionMode(ListSelectionModel.SINGLE_INTERVAL_SELECTION);
+		commandHistoryList.addListSelectionListener(new ListSelectionListener() {
+            @Override
+            public void valueChanged(ListSelectionEvent e) {
+
+                if (e.getValueIsAdjusting() == false) {
+
+                    if (commandHistoryList.getSelectedIndex() != -1) {
+                        int index = commandHistoryList.getSelectedIndex();
+                        List<DriverCommand> commands = CommandHistory.getCommandsFromList(index);
```

Comment:
This and the following lines should be extracted from view (CommandManagerWindow). Unnecessary dependencies with DriverCommand (and maybe CommandHistory).